### PR TITLE
Add JSON lockfile schema and validation function

### PR DESCRIPTION
**Pull request by jrdnbradford**
Here's my first go at issue #1877.

## Why Use [JSON Schema](https://json-schema.org/):

- Provides an easy, declarative way to document `renv.lock`
- Integrates with text editors to provide autocompletion, hover help, and validation
- Allows automated validation in CI, builds, tests, etc for R applications

## General Decisions

- A relatively relaxed schema to prevent false negatives, with the option for users to pass their own schema that fits their personal requirements. Additional fields are allowed.
- Heavy on positive and negative `testthat` tests, so that we can verify updates to the schema
- The function wraps [`jsonvalidate::json_validate()`](https://docs.ropensci.org/jsonvalidate/) and takes/passes many of its parameters there. I have not implemented all of them, but we can add if you’d like.
- Ease of use. Calling without parameters should cover most cases: `renv::lockfile_validate()`

## Questions

- Do I have all the fields `renv` knows about? 
- Do I have `required` set _only_ for the fields that are absolutely necessary?
- I’ve used `description`s from `renv` documentation whenever possible, but had to make some inferences. I myself have never used anything that added `Bioconductor`, `Python`, or a few other fields. These should be checked.

### CODE REVIEW

File: R/lockfile-validate.R [hunk 1]
```diff
@@ -0,0 +1,75 @@
+
+#' Validate the renv lockfile against a schema
```
Comment (kevinushey) [inline R/lockfile-validate.R:2]:
```suggestion
#' Validate an renv lockfile against a schema
```

File: R/lockfile-validate.R [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+  project <- renv_project_resolve(project)
+  lockfile <- lockfile %||% renv_lockfile_path(project = project)
+  schema <- schema %||% system.file("schema",
+                                    "draft-07.renv.lock.schema.json",
+                                    package = "renv",
+                                    mustWork = TRUE)
+
+  # "ajv" engine required for schema specifications later than draft-04
+  jsonvalidate::json_validate(lockfile,
```
Comment (kevinushey) [inline R/lockfile-validate.R:68]: Could we use either add `jsonvalidate` to the Suggests file of the package DESCRIPTION file, or do something like https://github.com/rstudio/renv/blob/5d0d52c395e569f7f24df4288d949cef95efca4e/R/dependencies.R#L1664-L1683 and better handle the presence (or absence) of the `jsonvalidate` package?

**Review by kevinushey — APPROVED** [review]
Overall LGTM -- thank you!

File: R/lockfile-validate.R (hunk 2 above)
Comment (jrdnbradford) [reply R/lockfile-validate.R:68]: Added to the `DESCRIPTION`'s `Suggests`.

**Comment by jrdnbradford** [comment]
Updates have been pushed. 

Presuming this gets merged, feel free to @ me on issues, bugs, etc on this. Schemas and their validation is always a balancing act, and the R community at large may need different tradeoffs. 

Closes #1877.

**Comment by kevinushey** [comment]
A huge thanks again for putting this together!

**Pull request merged**